### PR TITLE
rustc_metadata: Some reorganization of the module structure

### PR DESCRIPTION
**Pull request by petrochenkov**
The new structure of `rustc_metadata` (or rather its parts affected by the refactoring) is
```
├── lib.rs
└── rmeta
    ├── decoder
    │   └── cstore_impl.rs
    ├── decoder.rs
    ├── encoder.rs
    ├── mod.rs
    └── table.rs
```

(`schema` is renamed to `rmeta`.)

The code inside `rmeta` is pretty self-contained, so we can now privatize almost everything in this module instead of using `pub(crate)`  which was necessary when all these modules accessed their neighbors in the old flat structure.

`encoder` and `decoder` work with structures defined by `rmeta`.
`table` is a part of `rmeta`.
`cstore_impl` actively uses decoder methods and exposes their results through very few public methods (`provide` and `_untracked` methods).

r? @eddyb @spastorino

### CODE REVIEW

**Comment by eddyb** [comment]
I think I understand what the intent is, but I feel like "schema" stops making sense when it's more than just a bunch of type definitions.

I can think of two ways to improve on the naming (neither that great):
* rename `librustc_metadata` to `librustc_cross_crate` and `schema` to `metadata`
* *or* just rename `schema` to `rmeta` (on the basis that it's the "file format name")

I don't have a strong preference for having all the data types in their own module named `schema` (or if there is a good reason to do so, I've forgotten it), so your structure seems fine.

**Comment by petrochenkov** [comment]
Forgot to cc the related PRs - #66001, #66024, #66028.

**Comment by petrochenkov** [comment]
@eddyb 
I've renamed `schema` to `rmeta` and moved it from `rmeta.rs` to `rmeta/mod.rs`.

**Review by spastorino — APPROVED** [review]

**Comment by spastorino** [comment]
Seems good to me. Leaving the final call to @eddyb

File: src/librustc_metadata/rmeta/decoder/cstore_impl.rs [hunk 1]
```diff
@@ -529,6 +528,6 @@ impl CrateStore for cstore::CStore {
 
     fn metadata_encoding_version(&self) -> &[u8]
     {
-        schema::METADATA_HEADER
+        rmeta::METADATA_HEADER
```
Comment (eddyb) [inline src/librustc_metadata/rmeta/decoder/cstore_impl.rs]: Heh I just noticed the name of the method is kind of out of sync with what it's actually returning. No need to change it now though.

**Review by eddyb — APPROVED** [review]

**Pull request merged**